lib/prime.rb

```diff
@@ -132,7 +209,8 @@ def method_added(method) # :nodoc:
   #   Upper bound of prime numbers. The iterator stops after it
   #   yields all prime numbers p <= +ubound+.
   #
-  def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  #def each(ubound = nil, generator = EratosthenesGenerator.new, &block)
+  def each(ubound = nil, generator = NextPrimeGenerator.new, &block)
```

Comment:
The following changes makes the code shorter, more readable, and at lease twice as fast (for very large numbers), and is adapted from my `primes-utils` gem.  See PRIMES-UTILS HANDBOOK for math and code details.  
https://www.scribd.com/doc/266461408/Primes-Utils-Handbook
---------------------------------------------------------------------------
require 'openssl'

class Integer
  # Returns true if +self+ passes Miller-Rabin Test on +b+
  def miller_rabin_test(b)
    return self >= 2 if self <= 3 
    return false unless [1,5].include?(self % 6)
    n = d = self - 1
    d >>= 1 while d.even?
    y = b.to_bn.mod_exp(d, self)
    while d != n && y != n && y != 1
      y = y.to_bn.mod_exp(2, self)
      d <<= 1
    end
    return false if y != n && d.even?
    true
  end
  # Returns true if +self+ is a prime number, else returns false.
  def prime?
    return true if [2, 3, 5, 7].include? self
    return false unless self > 1 and 210.gcd(self % 210) == 1
    Prime::A014233.each do |pair|
      ix, mx = pair
      return false unless miller_rabin_test(ix)
      break if self < mx
    end
    return miller_rabin_test(41)
  end
end